sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2801,6 +2801,11 @@ class Analyzer(override val catalogManager: CatalogManager)
           p
         }
 
+      case g @ Generate(GeneratorOuter(generator), _, _, _, _, _) =>
+        g.copy(generator = generator,
```

Comment:
nit: `g.copy(generator = generator, outer = true)`

How about queries like this?  `select * from values 1, 2 lateral view outer explode_outer(array()) as b;`

Reply:
Thanks for reviewing!

>nit: g.copy(generator = generator, outer = true)

Ah yes, I will make the change.

>How about queries like this? select * from values 1, 2 lateral view outer explode_outer(array()) as b;

On master:
```
spark-sql> select * from values 1, 2 lateral view outer explode_outer(array()) as b;
Error in query: Column 'b' does not exist. Did you mean one of the following? [col1]; line 1 pos 26;
'Project [*]
+- 'Generate generatorouter(explode(array())), true, as, ['b]
   +- LocalRelation [col1#0]

spark-sql> 
```
With the PR:
```
spark-sql> select * from values 1, 2 lateral view outer explode_outer(array()) as b;
1	NULL
2	NULL
Time taken: 3.734 seconds, Fetched 2 row(s)
spark-sql> 
```
I can add it to the test.